src/main/java/teammates/storage/api/CoursesDb.java

```diff
@@ -65,6 +65,16 @@ public CourseAttributes getCourse(String courseId) {
         return new CourseAttributes(c);
     }
     
+    public List<CourseAttributes> getCourses(List<String> courseIds) {
+        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseIds);
+        List<Course> courses = getCourseEntities(courseIds);
+        List<CourseAttributes> courseAttributes = new ArrayList<CourseAttributes>();
+        for (Course c: courses) {
+            courseAttributes.add(new CourseAttributes(c));
+        }
+        return courseAttributes;
```

Comment:
Add a TODO here to extract a method `CourseAttributes.convert(List<Course>): List<CourseAttributes>`